UOP1_Project/Assets/Scripts/Sound/SoundEmitter.cs

```diff
@@ -76,4 +92,10 @@ public float LastUseTimestamp()
 	{
 		return _lastUseTimestamp;
 	}
+
+	IEnumerator FinishedPlaying(float clipLength)
+	{
+		yield return new WaitForSeconds(clipLength);
+		SoundManager.Instance.Pool.Return(this);
+	}
```

Comment:
This logic should be contained within the SoundEmitter's return sequence, rather than conditionally run if it's not a looping sound. Also, ideally, the pool object would not know about it's containing pool. I still think SoundEmitters _themselves_ should be decentralized to the users, with SoundManager as a provider. But if you're looking to keep it entirely decentralized, you can have SoundManager Return the SoundEmitter right after calling PlaySound().